auction-app/src/main/java/com/auctionapp/service/ProductService.java

```diff
@@ -1,23 +1,41 @@
 package com.auctionapp.service;
 
+import com.auctionapp.model.product.EProductCategory;
 import com.auctionapp.model.product.Product;
+import com.auctionapp.model.product.ProductDTO;
+import com.auctionapp.model.user.User;
 import com.auctionapp.repository.ProductRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
-
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
 @Service
 public class ProductService {
 
     private final ProductRepository productRepository;
+    private final UserService userService;
 
-    public ProductService(ProductRepository productRepository) {
+    public ProductService(ProductRepository productRepository, UserService userService) {
         this.productRepository = productRepository;
+        this.userService = userService;
     }
 
-    public void createProduct(Product product) {
+    public ResponseEntity<String> createProduct(Product product) {
+        if (product.getId() != null && product.getId() != 0) {
+            return ResponseEntity.badRequest().body("Invalid product ID");
+        }
+        //productRepository.save(product);
```

Comment:
as save is lower, probably this commented code is not needed

Reply:
Deleted